developers_chamber/qa/base.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import subprocess
+
+import click
+from git import Repo
+
+
+LOGGER = logging.getLogger()
+
+
+class QAError(Exception):
+    """
+    Generic QA exception that also carries command output.
+    """
+    def __init__(self, msg, output):
+        super().__init__(msg)
+        self.output = output.strip()
+
+
+class QACheck:
+    """
+    Base class for a quality assurance check.
+
+    Arguments:
+        name: Name of the check.
+    """
+    name = None
+
+    def _get_repo(self):
+        """
+        Returns the repo object.
+        """
+        return Repo('.')
+
+    def _get_default_branch(self):
+        """
+        Returns default branch of the repo.
+        """
+        return self._run_command("git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@'")
+
+    def _get_diffs(self):
+        """
+        Returns Git diffs against default branch.
+        """
+        repo = self._get_repo()
+        target_branch = getattr(repo.remotes.origin.refs, self._get_default_branch())
```

Comment:
I need to compare against remote branch, since Circle CI hard resets default branch to the PR branch (I absolutelly don't understand why).